### PR TITLE
CFITSIO - NASA Library for manipulating FITS files: new recipe

### PR DESCRIPTION
**Pull request by theclue**
CFITSIO is a public domain library for reading and writing files in FITS format, which is quite common in astronomy. This library is the first requisite for porting more astronomy and astrophysics software from NASA. Haikuporting is still an obscure art to me, but I'm trying to improve as much as possible.

- configure/make go ok outside hp
- while in haikuporter, the build stucks on 'waiting for build package cfitsio-3.47-1 to be activated' when adding `cmd:awk` to the `BUILD_PREREQUIRES `block.

I can see no more messages so I've no idea on where to start debugging this error.

### CODE REVIEW

**Comment by theclue** [comment]
ok, solved most issues. I'm going to update the pull request

**Comment by korli** [comment]
Please check spaces vs tabs.

**Comment by theclue** [comment]
> Please check spaces vs tabs.

Done. I guess I have to convert all tab-spaces to tabs and not vice-versa ;)

File: sci-libs/cfitsio/cfitsio-3.47.recipe [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+CHECKSUM_SHA256="418516f10ee1e0f1b520926eeca6b77ce639bed88804c7c545e74f26b3edf4ef"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64 ?arm ?ppc ?sparc"
+SECONDARY_ARCHITECTURES="x86 x86_64"
+
+PROVIDES="
+	cfitsio$secondaryArchSuffix = $portVersion
+	lib:libcfitsio$secondaryArchSuffix = $portVersion
+	"
+
```
Comment (korli) [inline sci-libs/cfitsio/cfitsio-3.47.recipe:19]: Please remove this blank Line.

File: sci-libs/cfitsio/cfitsio-3.47.recipe [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	haiku$secondaryArchSuffix
+	lib:libbz2$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	cfitsio${secondaryArchSuffix}_devel = $portVersion
+	devel:libcfitsio = $portVersion
+	"
+
```
Comment (korli) [inline sci-libs/cfitsio/cfitsio-3.47.recipe:30]: Please remove this blank Line.

File: sci-libs/cfitsio/cfitsio-3.47.recipe [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	lib:libcurl$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libbz2$secondaryArchSuffix
+	devel:libgfortran$secondaryArchSuffix
+	devel:libcurl$secondaryArchSuffix
+	"
+
```
Comment (korli) [inline sci-libs/cfitsio/cfitsio-3.47.recipe:43]: Please remove this blank Line.

File: sci-libs/cfitsio/cfitsio-3.47.recipe [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+BUILD_PREREQUIRES="
+	cmd:make
+	cmd:awk
+	cmd:gcc$secondaryArchSuffix
+	cmd:gfortran$secondaryArchSuffix
```
Comment (korli) [inline sci-libs/cfitsio/cfitsio-3.47.recipe:48]: cmd:make comes after cmd:gfortran

Comment (theclue) [reply sci-libs/cfitsio/cfitsio-3.47.recipe:44]: done....but why? alphabetical order?

**Pull request merged**